patches/minecraft/net/minecraft/command/CommandBase.java.patch

```diff
@@ -1,6 +1,22 @@
 --- ../src-base/minecraft/net/minecraft/command/CommandBase.java
 +++ ../src-work/minecraft/net/minecraft/command/CommandBase.java
-@@ -393,7 +393,7 @@
+@@ -96,6 +96,15 @@
+         return Collections.<String>emptyList();
+     }
+ 
++    private ICommand parentCommand = null;
```

Comment:
I simply rewrote the CommandTreeHelp, and I don't agree with other ideas. Adding new required class CommandChild just for CommandTreeBase wouldn't even work, since you can add other CommandTreeBase commands in it (thus, the name, tree) unless it itself is CommandChild. And I don't see how parameters would help either. I can make it "more immutable" but I don't see the point in that too - normal user just has to use addSubcommand and nothing else. If a mod wants to mess with commands then probably for a reason, like a permission mod

Reply:
Right, CommandTreeBase would extend CommandChild (or better named CommandNode). Whatever can require not  patching in a field, which should really never be a first solution.